apis/concierge/authentication/v1alpha1/types_tls.go.tmpl

```diff
@@ -3,9 +3,24 @@
 
 package v1alpha1
 
+// CABundleSource provides a source for CA bundle used for client-side TLS verification.
+type CABundleSource struct {
+	// Whether the CA bundle is being sourced from a kubernetes secret or a configmap.
+	// +kubebuilder:validation:Enum=Secret;ConfigMap
+	Kind string `json:"kind,omitempty"`
+	// Name of the secret or configmap from which to read the CA bundle.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+	// Key within the secret or configmap from which to read the CA bundle.
+	// +kubebuilder:validation:MinLength=1
+	Key string `json:"key,omitempty"`
+}
+
 // Configuration for configuring TLS on various authenticators.
 type TLSSpec struct {
 	// X.509 Certificate Authority (base64-encoded PEM bundle). If omitted, a default set of system roots will be trusted.
 	// +optional
 	CertificateAuthorityData string `json:"certificateAuthorityData,omitempty"`
+	// +optional
+	CertificateAuthorityDataSource CABundleSource `json:"certificateAuthorityDataSource,omitempty"`
```

Comment:
I think this should this be a pointer, because it is optional, right? Try installing your generated CRD (for one of the CRDs that uses this struct) into a Kind cluster and then play with creating resources of that type. Unless this field is a pointer, I fear that the validations that live on `CABundleSource` may not work correctly.

Reply:
Also, this field should have a comment, right? Otherwise it might look odd when you `kubectl explain` it.

Reply:
Yes, it makes sense to have the `CertificateAuthorityDataSource ` be a pointer. I will also add the missing comment.
I did install the CRD to catch any codegen errors from the template. But I will also create CRs of the type just to see how it looks.

Reply:
Updated the spec and also added some integration tests.